sot_hpp/ros_interface.py

```diff
@@ -104,7 +105,17 @@ def readQueue(self, req):
             answer = self.runCommand (cmd)
             print cmd
             print answer
-        return SetBoolResponse (True, "ok")
+        return SetIntResponse ()
+
+    def getSoTTime(self, req):
+        if self.supervisor is not None:
+            return GetIntResponse (self.supervisor.sotrobot.device.control.time)
+        else:
+            cmd = "supervisor.sotrobot.device.control.time"
+            answer = self.runCommand (cmd)
+            print cmd
+            print answer
+            return GetIntResponse (int(answer.result))
```

Comment:
I guess from this change that you intent to use it as follows:
```
t = getSoTTime()
readQueue (t + shift)
```
where `shift>= 1` is an integer.

I fear this two calls to ROS services will add an unpredictable delay between them. Either `shift` is set to a high value or it may happen that `t + shift` is not greater than the new SoT time when readQueue is executed and `readQueue` may run into trouble.

Wouldn't it be better to have only one function `readQueue` like the following ?
```python
def readQueue(shift)
  t = getSoTTime()
  self.supervisor.readQueue(t + shift)
```

There should not be any delay issues. Moreover, I would rather say that exposing SoT time is not desirable.